src/target.rs

```diff
@@ -38,14 +39,25 @@ pub enum Target<F: Field> {
 impl<Fp: Field> Target<Fp> {
     pub fn convert<Fq: Field>(self) -> Target<Fq> {
         match self {
+            Target::PublicInput(pi) => Target::PublicInput(pi.convert()),
             Target::VirtualTarget(v) => Target::VirtualTarget(v),
             Target::Wire(w) => Target::Wire(w),
             _ => unreachable!(),
         }
     }
+
     pub fn convert_slice<Fq: Field>(s: &[Self]) -> Vec<Target<Fq>> {
         s.iter().map(|t| t.convert()).collect()
     }
+
+    pub fn index(&self) -> usize {
+        match self {
+            Target::PublicInput(pi) => pi.index,
+            Target::VirtualTarget(v) => v.index,
+            Target::Wire(w) => w.gate,
+            Target::_Field(_, _) => unreachable!()
+        }
+    }
```

Comment:
This is currently used only to get the the public input index in `plonk_recursion.rs`. I hesitated to make it panic when `Target` isn't a public input. What do you think?